libraries/AP_RAMTRON/AP_RAMTRON.cpp

```diff
@@ -92,6 +95,30 @@ bool AP_RAMTRON::init(void)
         return false;
     }
 
+    if (has_backup()) {
+        ramtron_logdata = new uint8_t[get_size()];
+        ramtron_logdata_len = get_size();
+
+        if (ramtron_logdata != nullptr) {
+            //enter a random fill value, to tell us if 
+            //we failed read bytes from fram
+            memset(ramtron_logdata, 0xAB, get_size());
+            uint16_t log_offset = 0;
+            while (true) {
+                //read 256 bytes at a time
+                if (!_read_data(log_offset, &ramtron_logdata[log_offset], 256)) {
+                    break;
+                } else {
+                    log_offset+=256;
+                }
+                if (log_offset >= get_size()) {
+                    break;
+                }
+            }
+        }
+        hal.scheduler->register_io_process(FUNCTOR_BIND_MEMBER(&AP_RAMTRON::data_integrity_tracker, void));
```

Comment:
We probably ought not be doing this if we fail to get our integrity data successfully.

Reply:
This integrity tracker is independent to the data we collected above, with integrity tracker we  only record data to the second half of the storage and calculate crc.